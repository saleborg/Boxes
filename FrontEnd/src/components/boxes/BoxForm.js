import React from "react";
import PropTypes from "prop-types";
import TextInput from "../common/TextInput";
import SelectInput from "../common/SelectInput";
import { SketchPicker } from "react-color";

const BoxForm = ({
  box,
  color,
  onSave,
  onChange,
  onChangeColor,
  saving = false,
  errors = {},
  handleColorPicker,
  showPicker = false,
}) => {
  return (
    <form onSubmit={onSave} name="submitform">
      <h2>Add Box</h2>
      {errors.onSave && (
        <div className="alert alert-danger" role="alert">
          {errors.onSave}
        </div>
      )}
      <TextInput
        id="name"
        name="name"
        label="Name"
        value={box.name}
        onChange={onChange}
        error={errors.name}
      />
      <TextInput
        keyboardType="numeric"
        name="weight"
        label="Weight"
        value={box.weight ? String(box.weight) : ""}
        onChange={onChange}
        error={errors.weight}
      />

      <input
        type="button"
        value="Click to show Color picker"
        onClick={handleColorPicker}
        error={errors.color}
      />

      {showPicker && (
        <SketchPicker
          id="picker"
          color={color}
          onChange={onChangeColor}
          onChangeComplete={onChangeColor}
        />
      )}

      {errors.color && <div className="alert alert-danger">{errors.color}</div>}
      <SelectInput
        name="country"
        label="Country"
        value={box.country || ""}
        defaultOption="Select Country"
        options={countries.map((country) => ({
          value: country,
          text: country,
        }))}
        onChange={onChange}
        error={errors.country}
      />
      <button type="submit" disabled={saving} className="btn btn-primary">
        {saving ? "Saving..." : "Save"}
      </button>
    </form>
  );
};

BoxForm.propTypes = {
  box: PropTypes.object.isRequired,
  color: PropTypes.object.isRequired,
  onSave: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  errors: PropTypes.object,
  saving: PropTypes.bool,
  onChangeColor: PropTypes.func.isRequired,
  handleColorPicker: PropTypes.func.isRequired,
  showPicker: PropTypes.bool,
};

const countries = ["Sweden", "China", "Brazil", "Australia"];

export default BoxForm;
