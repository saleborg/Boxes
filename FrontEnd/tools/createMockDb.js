/* eslint-disable no-console */
const fs = require("fs");
const path = require("path");
const mockData = require("./mockData");

const { boxes } = mockData;
const data = JSON.stringify({ boxes });
const filepath = path.join(__dirname, "db.json");

fs.writeFile(filepath, data, function (err) {
  err ? console.log(err) : console.log("Mock DB created.");
});
