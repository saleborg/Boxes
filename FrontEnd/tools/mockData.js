const boxes = [
  {
    idboxes: 39,
    name: "jhg",
    shippingcost: 72.8,
    color: "100,100,100",
    weight: 56,
  },
  {
    idboxes: 42,
    name: "Anna",
    shippingcost: 2012.3999999999999,
    color: "245,18,18",
    weight: 234,
  },
  {
    idboxes: 43,
    name: "fsd",
    shippingcost: 2332.8,
    color: "180,84,134",
    weight: 324,
  },
];
const newBox = {
  idboxes: "",
  name: "",
  shippingcost: "",
  color: "",
  weight: "",
};

module.exports = {
  newBox,
  boxes,
};
