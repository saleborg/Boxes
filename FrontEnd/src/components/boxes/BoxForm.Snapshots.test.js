import React from "react";
import BoxForm from "./BoxForm";
import renderer from "react-test-renderer";
import { boxes } from "../../../tools/mockData";

it("sets submit button label 'Saving...' when saving is true", () => {
  const box = boxes[0];
  const color = { r: 0, g: 0, b: 0 };
  const tree = renderer.create(
    <BoxForm
      box={box}
      onChange={jest.fn}
      onSave={jest.fn}
      saving
      color={color}
      handleColorPicker={jest.fn}
      onChangeColor={jest.fn}
    />
  );

  expect(tree).toMatchSnapshot();
});

it("sets submit button label 'Save' when saving is false", () => {
  const box = boxes[0];
  const color = { r: 0, g: 0, b: 0 };
  const tree = renderer.create(
    <BoxForm
      box={box}
      onChange={jest.fn}
      onSave={jest.fn}
      saving={false}
      color={color}
      handleColorPicker={jest.fn}
      onChangeColor={jest.fn}
    />
  );

  expect(tree).toMatchSnapshot();
});
