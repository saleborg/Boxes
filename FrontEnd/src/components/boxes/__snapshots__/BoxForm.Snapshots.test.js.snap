// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sets submit button label 'Save' when saving is false 1`] = `
<form
  name="submitform"
  onSubmit={[Function]}
>
  <h2>
    Add Box
  </h2>
  <div
    className="form-group"
  >
    <label
      htmlFor="name"
    >
      Name
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        id="name"
        name="name"
        onChange={[Function]}
        type="text"
        value="jhg"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="weight"
    >
      Weight
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="weight"
        onChange={[Function]}
        type="text"
        value="56"
      />
    </div>
  </div>
  <input
    onClick={[Function]}
    type="button"
    value="Click to show Color picker"
  />
  <div
    className="form-group"
  >
    <label
      htmlFor="country"
    >
      Country
    </label>
    <div
      className="field"
    >
      <select
        className="form-control"
        name="country"
        onChange={[Function]}
        value=""
      >
        <option
          value=""
        >
          Select Country
        </option>
        <option
          value="Sweden"
        >
          Sweden
        </option>
        <option
          value="China"
        >
          China
        </option>
        <option
          value="Brazil"
        >
          Brazil
        </option>
        <option
          value="Australia"
        >
          Australia
        </option>
      </select>
    </div>
  </div>
  <button
    className="btn btn-primary"
    disabled={false}
    type="submit"
  >
    Save
  </button>
</form>
`;

exports[`sets submit button label 'Saving...' when saving is true 1`] = `
<form
  name="submitform"
  onSubmit={[Function]}
>
  <h2>
    Add Box
  </h2>
  <div
    className="form-group"
  >
    <label
      htmlFor="name"
    >
      Name
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        id="name"
        name="name"
        onChange={[Function]}
        type="text"
        value="jhg"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="weight"
    >
      Weight
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="weight"
        onChange={[Function]}
        type="text"
        value="56"
      />
    </div>
  </div>
  <input
    onClick={[Function]}
    type="button"
    value="Click to show Color picker"
  />
  <div
    className="form-group"
  >
    <label
      htmlFor="country"
    >
      Country
    </label>
    <div
      className="field"
    >
      <select
        className="form-control"
        name="country"
        onChange={[Function]}
        value=""
      >
        <option
          value=""
        >
          Select Country
        </option>
        <option
          value="Sweden"
        >
          Sweden
        </option>
        <option
          value="China"
        >
          China
        </option>
        <option
          value="Brazil"
        >
          Brazil
        </option>
        <option
          value="Australia"
        >
          Australia
        </option>
      </select>
    </div>
  </div>
  <button
    className="btn btn-primary"
    disabled={true}
    type="submit"
  >
    Saving...
  </button>
</form>
`;
